uint32 size
float64[] means
float64[] cov_mat
